rules/no-invalid-remove-event-listener.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const methodSelector = require('./utils/method-selector');
+
+const MESSAGE_ID = 'no-invalid-remove-event-listener';
+const messages = {
+	[MESSAGE_ID]:
+		'Only call removeEventListener with the same reference to a function passed to addEventListener.'
+};
+
+const removeEventListenerSelector = methodSelector({
+	name: 'removeEventListener'
+});
+
+/** @param {import('eslint').Rule.RuleContext} context */
+const create = context => {
+	return {
+		[removeEventListenerSelector]: node => {
+			const secondArgument = node.arguments[1];
+			if (!secondArgument) {
```

Comment:
This is rarely seen, 

```
foo.removeEventListener(...args, getListener())
```

but let's make sure the first argument is not a `SpreadElement`, you can do this by changing 

```
const removeEventListenerSelector = methodSelector({
	name: 'removeEventListener'
});
```js

to 

```js
const removeEventListenerSelector = methodSelector({
	name: 'removeEventListener',
	min: 2
});
```